src/vs/workbench/contrib/terminal/node/terminalProfiles.ts

```diff
@@ -0,0 +1,273 @@
+						}
+					} else {
+						let resolved = variableResolver?.resolve(workspaceFolder, customProfile.path);
+						if (resolved) {
+							profile = detectedProfiles?.find(profile => profile.path === resolved);
+						} else if (statProvider) {
+							// used by tests
+							resolved = customProfile.path;
+						} else {
+							logService?.trace(`Could not resolve path ${customProfile.path} in workspace folder ${workspaceFolder}`);
+						}
+						if (!profile) {
+							logService?.trace(`Could not detect path ${resolved}`);
+						}
+					}
+					if (profile) {
+						if (customProfile.args) {
+							validProfiles?.push({ profileName: profileKey, path: profile.path, args: customProfile.args });
+						} else {
+							validProfiles?.push({ profileName: profileKey, path: profile.path });
+						}
+					}
+				} else if ((value as ITerminalProfileGenerator).generator) {
+					// generator
+					let generatorKey = (value as ITerminalProfileGenerator).generator;
+					const profile = detectedProfiles?.find(profile => profile.profileName === generatorKey.toString());
+					if (profile) {
+						validProfiles?.push({ profileName: profileKey, path: profile.path, args: profile.args });
+					} else {
+						logService?.trace(`No generator with key ${generatorKey}`);
+					}
+				} else {
+					logService?.trace(`Entry in terminal.profiles.windows is not of type ITerminalExecutable or ITerminalProfileGenerator`, profileKey, value);
+				}
+			}
+		}
+	} else {
+		logService?.trace(`No detected profiles ${JSON.stringify(detectedProfiles)} or ${JSON.stringify(configProfiles)}`);
+	}
+	return validProfiles;
+}
+
+async function getPowershellProfiles(): Promise<IPotentialTerminalProfile[]> {
+	let profiles: IPotentialTerminalProfile[] = [];
+	// Add all of the different kinds of PowerShells
+	for await (const pwshExe of enumeratePowerShellInstallations()) {
+		profiles.push({ profileName: pwshExe.displayName, paths: [pwshExe.exePath] });
+	}
+	return profiles;
+}
+
+async function getWslProfiles(wslPath: string, detectWslProfiles?: boolean): Promise<IPotentialTerminalProfile[]> {
+	let profiles: IPotentialTerminalProfile[] = [];
+	if (detectWslProfiles) {
+		const distroOutput = await new Promise<string>(r => cp.exec('wsl.exe -l', (err, stdout) => err ? console.trace('problem occurred when getting wsl distros', err) : r(stdout)));
+		if (distroOutput) {
+			let regex = new RegExp(/[\r?\n]/);
+			let distroNames = Buffer.from(distroOutput).toString('utf8').split(regex).filter(t => t.trim().length > 0 && t !== '');
+			// don't need the Windows Subsystem for Linux Distributions header
+			distroNames.shift();
+			for (const distroName of distroNames) {
+				let s = '';
+				let counter = 0;
+				for (const c of Array.from(distroName)) {
+					if (counter % 2 === 1) {
+						// every other character is junk / a rectangle
+						s += c;
+					}
+					counter++;
+				}
+				if (s.endsWith('(Default)')) {
```

Comment:
Should be easier, don't even need to check if it ends with imo

`s.remove(' (Default)'`

Reply:
remove doesn't exist on type string